vernac/vernac_classifier.ml

```diff
@@ -55,21 +55,23 @@ let classify_vernac e =
     (* Univ poly compatibility: we run it now, so that we can just
      * look at Flags in stm.ml.  Would be nicer to have the stm
      * look at the entire dag to detect this option. *)
-    | VernacSetOption (_, l,_)
-      when CList.exists (CList.equal String.equal l)
-        options_affecting_stm_scheduling ->
-       VtSideff ([], VtNow)
-    | VernacBeginSection {v=id} -> VtSideff ([id], VtLater)
+    | VernacSetOption (_, l,_) ->
+      let now =
+        if CList.exists (CList.equal String.equal l) options_affecting_stm_scheduling
+        then VtNow
+        else VtLater
+      in
+       VtSideff ([], now)
+    | VernacBeginSection {v=id} -> VtSideff ([id], VtNow)
     | VernacChdir _ | VernacExtraDependency _
-    | VernacSetOption _ -> VtSideff ([], VtLater)
     (* (Local) Notations have to disappear *)
     | VernacEndSegment _ -> VtSideff ([], VtNow)
     (* Modules with parameters have to be executed: can import notations *)
     | VernacDeclareModule (exp,{v=id},bl,_)
     | VernacDefineModule (exp,{v=id},bl,_,_) ->
-        VtSideff ([id], if bl = [] && exp = None then VtLater else VtNow)
+        VtSideff ([id], VtNow)
     | VernacDeclareModuleType ({v=id},bl,_,_) ->
-        VtSideff ([id], if bl = [] then VtLater else VtNow)
+        VtSideff ([id], VtNow)
     (* These commands alter the parser *)
```

Comment:
Why? Can you explain?

Reply:
the idea is that we already did classification for synterp / synpure so we can reuse it here
but maybe that's incorrect
this PR probably isn't worth finishing anyway